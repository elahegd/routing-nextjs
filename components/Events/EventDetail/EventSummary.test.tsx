import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import EventSummary from "./EventSummary";

describe("EventSummary", () => {
  it("renders title correctly", () => {
    const title = "Test Title";
    const { getByText } = render(<EventSummary title={title} />);
    const headerElement = getByText(title);
    expect(headerElement).toBeInTheDocument();
  });

  it("has correct class name", () => {
    const title = "Test Title";
    const { container } = render(<EventSummary title={title} />);
    const sectionElement = container.querySelector("section");
    expect(sectionElement).toHaveClass("summary");
  });

  it("matches snapshot", () => {
    const title = "Test Title";
    const { asFragment } = render(<EventSummary title={title} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
