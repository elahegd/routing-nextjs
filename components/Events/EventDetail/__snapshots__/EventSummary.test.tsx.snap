// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventSummary matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="summary"
  >
    <h1>
      Test Title
    </h1>
  </section>
</DocumentFragment>
`;
